questions/blockdelay.md

```diff
@@ -13,49 +13,144 @@ Using repeaters with command blocks is considered bad practice, and is often imp
+
+    # 1.13 - 1.20.4
+    give @s bat_spawn_egg{EntityTag:{id:"minecraft:area_effect_cloud",Tags:["delay"],Duration:100,Radius:0f,WaitTime:0}}
+    
+    # 1.20.5+
+    give @s bat_spawn_egg[entity_data={id:"minecraft:area_effect_cloud",Tags:["delay"],Duration:100,Radius:0f,WaitTime:0}]
+
+_Note: You can use any spawn_egg, but not just `bat_spawn_egg`._
+
+    # Command block / tick function
+    execute at @e[type=area_effect_cloud,tag=delay,nbt={Age:99}] run summon zombie
+
+This is a simple way to execute any command at a specified position once with a specified delay.
+
+### Marker
+
+If you need to execute a command not only once, but every 5 seconds, for example, at specific location, then you can use the [marker entity](https://minecraft.wiki/w/Marker) (1.17+) for this. If you are on an earlier version use the invisible armor_stand.
+
+    # Summon
+    summon marker ~ ~ ~ {Tags:["delay"]}
+    
+    # Spawn egg
+    ## 1.13 - 1.20.4
+    give @s bat_spawn_egg{EntityTag:{id:"minecraft:marker",Tags:["delay"]}}
+    
+    ## 1.20.5+
+    give @s bat_spawn_egg[entity_data={id:"minecraft:marker",Tags:["delay"]}]
 
-That means that if you have those 3 commands repeating somewhere in your world, the command `/summon area_effect_cloud X Y Z {Age:-##,Particle:"take",Tags:[delay]}` can be used to activate the command block at (X, Y, Z) in ## ticks. An example, that'll activate the block at (10, 64, 30) in 70 ticks:
+In addition to the marker, you need to use a scoreboard timer, which each tick will add 1 to the score marker.
 
-    summon area_effect_cloud 10 64 30 {Age:-70,Particle:"take",Tags:["delay"]}
+    # Command blocks / tick function
+    scoreboard players add @e[type=marker,tag=delay] timer 1
+    execute as @e[type=marker,tag=delay,scores={delay=100..}] at @s store success score @s delay run summon zombie
 
 ### Scoreboard
 
 For a scoreboard timer you can have a repeating commandblock somewhere that's counting up/down in a particular scoreboard objective and then use `execute if score` in the commandblock that should have the delay. You can either use individual player scores (recommended for player dependant events/delays) or "[fake player](/wiki/questions/fakeplayer)" scores (set "fake" values for player names, recommended for player independant delays).
 
-    # the repeating commandblock
+    # Setup
+    scoreboard objectives add timer dummy
+    
+    # The repeating command_block
     scoreboard players add @a timer 1
 
-    # the commandblock that should run for every player with a timer score of 100
-    execute as @a[scores={timer=100}] run say this command has 5 seconds delay
+    # The command_block that should run for every player with a timer score of 100
+    execute as @a[scores={timer=100}] run say This command has 5 seconds delay.
+    scoreboard players reset @a[scores={timer=100..}] timer
 
-    # alternatively for a fake player timer
-    scoreboard players add $FakePlayer timer 1
```

Comment:
With the addition of macros in 1.20.2 that also use $, this may now confuse some into thinking that macros are being used somewhere, especially if it is used in other places that actually use macros. I understand that there are no macros here, but still.

Reply:
Yes, that is a good point. I will update the fake player article with this information so we have the info in one central place.